packages/SwingSet/src/kernel/vatManager/manager-subprocess-xsnap.js

```diff
@@ -72,19 +70,14 @@ export function makeXsSubprocessFactory({
       return doSyscall(vatSyscallObject);
     }
 
-    /** @type { (vref: unknown, count: number) => void } */
-    function vatDecref(vref, count) {
-      decref(vatID, vref, count);
-    }
-
     /** @type { (item: Tagged) => unknown } */
     function handleUpstream([type, ...args]) {
       parentLog(vatID, `handleUpstream`, type, args.length);
       switch (type) {
         case 'syscall': {
           parentLog(vatID, `syscall`, args[0], args.length);
-          const [scTag, ...vatSyscallArgs] = args;
-          return handleSyscall([scTag, ...vatSyscallArgs]);
+          const vatSyscallObject = args[0];
+          return handleSyscall(vatSyscallObject);
```

Comment:
A `Tagged` is a non-empty array. The only typesafe way I know to establish that an `unknown` is an `Array` is to assert that it is. Then, to convince `tsc` that we're providing a non-empty array to `handleSyscall`, a technique I have used is to destructure the first item...

```suggestion
          assert(Array.isArray(args[0]), 'vatSyscallObject must be an Array');
          const [scTag, ...vatSyscallArgs] = args[0];
          return handleSyscall([scTag, ...vatSyscallArgs]);
```